server/controllers/file.controller.js

```diff
@@ -77,44 +99,68 @@ function deleteChild(files, parentId, id) {
   });
 }
 
-export function deleteFile(req, res) {
-  Project.findById(req.params.project_id, (err, project) => {
+export async function deleteFile(req, res) {
+  try {
+    let project = await Project.findById(req.params.project_id);
     if (!project) {
-      res.status(404).send({ success: false, message: 'Project does not exist.' });
+      res
+        .status(404)
+        .send({ success: false, message: 'Project does not exist.' });
+      return;
     }
     if (!project.user.equals(req.user._id)) {
-      res.status(403).send({ success: false, message: 'Session does not match owner of project.' });
+      res.status(403).send({
+        success: false,
+        message: 'Session does not match owner of project.'
+      });
       return;
     }
-
-    // make sure file exists for project
-    const fileToDelete = project.files.find(file => file.id === req.params.file_id);
+    const fileToDelete = project.files.find(
+      file => file.id === req.params.file_id
+    );
     if (!fileToDelete) {
-      res.status(404).send({ success: false, message: 'File does not exist in project.' });
+      res
+        .status(404)
+        .send({ success: false, message: 'File does not exist in project.' });
       return;
     }
 
+    // make sure file exists for project
     const idsToDelete = getAllDescendantIds(project.files, req.params.file_id);
     deleteMany(project.files, [req.params.file_id, ...idsToDelete]);
-    project.files = deleteChild(project.files, req.query.parentId, req.params.file_id);
-    project.save((innerErr) => {
-      res.json(project.files);
-    });
-  });
+    project.files = deleteChild(
+      project.files,
+      req.query.parentId,
+      req.params.file_id
+    );
+    project = await project.save();
+    res.json(project.files);
+  } catch (err) {
+    // NOT SO GREAT ERROR HANDLING
+    console.log(err);
```

Comment:
We should probably respond with a 500 error here so that the API user always gets a response - though don't send the user the error